common/lib/client/realtimechannel.js

```diff
@@ -505,12 +498,12 @@ var RealtimeChannel = (function() {
 		switch(this.state) {
 			case 'attaching':
 				var err = new ErrorInfo('Channel attach timed out', 90000, 408);
-				this.setState('detached', err);
-				this.failPendingMessages(err);
+				this.notifyState('suspended', err);
+				this.failPendingMessages(err); // RTP5f
 				break;
 			case 'detaching':
 				var err = new ErrorInfo('Channel detach timed out', 90000, 408);
-				this.setState('attached', err);
+				this.notifyState('attached', err);
```

Comment:
Channel state is decoupled from connection state (in most cases unless it becomes suspended / failed).  As such, it's very plausible that an `ATTACH` / `DETACH` could be sent to realtime, actioned by realtime, but the client never receives the ack on the current transport.  In the case that the transport and thus connection becomes disconnected, when the ACK has not been received, the connection will then reestablish a transport and continue from where it left off, yet now the client library does not know if the channel is `ATTACHED` or `DETACHED`. We have discussed this before, and I would suggest that any pending channel state request is resent immediately when the connection becomes connected again. I don't see any downside to this, and given there is no serial number change when for channel state changes, there is simply no way for a client library to know if the ATTACH / DETACH request was actually accepted or not.